From 3fc4ebaf2eccb062f059a1f6ea229146454adf10 Mon Sep 17 00:00:00 2001
From: Corey Minyard <cminyard@mvista.com>
Date: Wed, 3 Feb 2016 09:19:28 -0600
Subject: [PATCH 3/3] ppc: Add kernel entry point to kernel crashdump info

For gdb to debug a relocated kernel, it needs to know the actual
point of kernel entry in AT_ENTRY in the .auxv note.  So add that
entry to the crashdump information so the kdump extraction tools
can get it.

Signed-off-by: Corey Minyard <cminyard@mvista.com>
---
 arch/powerpc/kernel/machine_kexec.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/powerpc/kernel/machine_kexec.c b/arch/powerpc/kernel/machine_kexec.c
index df5436e..356098f 100644
--- a/arch/powerpc/kernel/machine_kexec.c
+++ b/arch/powerpc/kernel/machine_kexec.c
@@ -65,6 +65,8 @@ void machine_kexec_cleanup(struct kimage *image)
 {
 }
 
+extern void _stext(void);
+
 void arch_crash_save_vmcoreinfo(void)
 {
 #ifndef CONFIG_PPC64
@@ -89,6 +91,10 @@ void arch_crash_save_vmcoreinfo(void)
 #ifndef CONFIG_NEED_MULTIPLE_NODES
 	VMCOREINFO_SYMBOL(contig_page_data);
 #endif
+#ifdef CONFIG_RELOCATABLE
+	vmcoreinfo_append_str("ADDRESS(%s)=%llx\n", "entry",
+			      (unsigned long long) _stext);
+#endif
 }
 
 /*
-- 
2.5.0

